### PR TITLE
Auth JS service

### LINKED ISSUE #367: Add auth service JS code
References user story #277 

We need some service functions that handle logging a user in, creating a new user, logging out, and getting the current user from the local storage. See this example here https://www.bezkoder.com/react-jwt-auth/

### PR DESCRIPTION
**Pull request by AshlynAdamson**
Closes #367

Getting auth service functions for the front end created. I used this example as I was developing, https://www.bezkoder.com/react-jwt-auth/. I followed the same idea for the page title stuff for the currentUser constant. There will most likely be changes needed here as more functionality gets integrated

### CODE REVIEW

File: BEIMA.Client/src/services/Authentication.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  setCurrentUser({});
+}
+
+/**
+ * Checks if the token is in localStorage and sets the currentUser constant
+ * This is used for if a user selects remember me, but closes their browser
+ * before the timeout occurs
+ * 
+ */
+export default function checkCurrentUser(){
```
Comment (josephmoore99) [inline BEIMA.Client/src/services/Authentication.js:69]:
```suggestion
export default function checkAndSetCurrentUser(){
```

File: BEIMA.Client/src/services/Authentication.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+      setCurrentUser(user);
+      return;
+    });
+  
+    return login.response;
+}
+
+/**
+ * Removed the token from the browser storage
```
Comment (josephmoore99) [inline BEIMA.Client/src/services/Authentication.js:48]:
```suggestion
 * Remove the token from the browser storage
```

**Comment by josephmoore99** [comment]
Do we have functionality on all the pages to check that we are logged in? Or I guess how would the frontend respond if we tried to access a page where we did not have full permissions on?

**Comment by AshlynAdamson** [comment]
> Do we have functionality on all the pages to check that we are logged in? Or I guess how would the frontend respond if we tried to access a page where we did not have full permissions on?

This will be done in tasks #369 and #370. If a user tries to access something they don't have access to, like the Users page, we can redirect them to the Devices page. Or if a user isn't logged in and tries to redirect using the url, it will redirect them to the login page

**Review by josephmoore99 — APPROVED** [review]

**Comment by AshlynAdamson** [comment]
I am closing this PR since I ended up changing the work I did here in this PR, https://github.com/cs481-ekh/s22-beima/pull/441,  and I'll just keep it all in one PR

**Pull request closed without merging**